### PR TITLE
Kh dev admin dashboard

### LINKED ISSUE #15: style items index
mobile-first styling for items index

### PR DESCRIPTION
**Pull request by kheppenstall**
-Create enum for role with admin (and tests)
-Feature helper method `#login_user(user)`
-Create admin dashboard route
-Admin authentication with login
-Admin authorization in `base_controller`
-Create `admin::base_controller` `#current_admin?`
-Namespace admin routes in `routes.rb`

Closes #15

Everyone should review. @akintner @tmikeschu @wlffann Definitely check out the file structure to and give feedback on that.

Admin dashboard still needs styling (added as a waffle card).

### CODE REVIEW

File: app/controllers/admin/base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  def require_admin
+    render file: "/public/404" unless current_admin?
+  end
+
+  def dashboard
+    render 'admin/dashboard'
+  end
+
+  def current_admin?
+    current_user && current_user.admin?
```
Comment (tmikeschu) [inline app/controllers/admin/base_controller.rb]: Should current_admin live in the application controller? Something to keep in mind as we build functionality.

File: spec/features/sessions/visitor_logs_in_spec.rb [hunk 2]
```diff
@@ -20,10 +20,9 @@
       user = create(:user)
 
       visit login_path
+      login_user(user)
```
Comment (tmikeschu) [inline spec/features/sessions/visitor_logs_in_spec.rb]: Nice refactor!

File: app/controllers/admin/base_controller.rb (hunk 1 above)
Comment (kheppenstall) [inline app/controllers/admin/base_controller.rb]: @tmikeschu Definitely I'll make that change.

**Comment by akintner** [comment]
@kheppenstall All this looks good. I agree with @tmikeschu  that eventually we might want to put current_admin in the application_controller, but I think that can wait for now.

**Comment by tmikeschu** [comment]
@kheppenstall  The logic looks clean, and I agree with the folder structure. Good to merge on @wlffann 's approval.

**Comment by kheppenstall** [comment]
Update to move `#current_admin?` to `ApplicationController`

**Comment by wlffann** [comment]
@kheppenstall Thanks for making those changes! Looks good now.

**Pull request merged**